### ISSUE TITLE
Cannot download JVM: temurin:17

### ISSUE THREAD

**Issue by fawdlstty** [VirtusLab/scala-cli#2861]
**Version(s)**
3.4.1

**Describe the bug**
can not use

**To Reproduce**
```
D:\projects_chisel>scala-cli chisel-example.scala --verbose
Error: java.lang.Exception: scala.build.errors.JvmDownloadError: Cannot download JVM: temurin:17
For more details, please see 'D:\projects_chisel\.scala-build\stacktraces\1713366276-7143304691416179470.log'
```

```
java.lang.Exception: scala.build.errors.JvmDownloadError: Cannot download JVM: temurin:17
  scala.cli.internal.CliLogger.exit(CliLogger.scala:145)
  scala.cli.commands.util.CommandHelpers$EitherBuildExceptionOps.orExit(CommandHelpers.scala:18)
  scala.cli.commands.run.Run$.runCommand(Run.scala:145)
  scala.cli.commands.default.Default.runCommand(Default.scala:61)
  scala.cli.commands.default.Default.runCommand(Default.scala:40)
  scala.cli.commands.ScalaCommand.run(ScalaCommand.scala:368)
  scala.cli.commands.ScalaCommand.run(ScalaCommand.scala:350)
  caseapp.core.app.CaseApp.main(CaseApp.scala:157)
  scala.cli.commands.ScalaCommand.main(ScalaCommand.scala:335)
  caseapp.core.app.CommandsEntryPoint.main(CommandsEntryPoint.scala:166)
  scala.cli.ScalaCliCommands.main(ScalaCliCommands.scala:125)
  scala.cli.ScalaCli$.main0(ScalaCli.scala:276)
  scala.cli.ScalaCli$.main(ScalaCli.scala:108)
  scala.cli.ScalaCli.main(ScalaCli.scala)
scala.build.errors.JvmDownloadError: Cannot download JVM: temurin:17
  scala.cli.commands.util.JvmUtils$.downloadJvm(JvmUtils.scala:81)
  scala.cli.commands.shared.SharedOptions.defaultJvmHome$lzyINIT1$1(SharedOptions.scala:532)
  scala.cli.commands.shared.SharedOptions.defaultJvmHome$1(SharedOptions.scala:533)
  scala.cli.commands.shared.SharedOptions.$anonfun$40(SharedOptions.scala:544)
  scala.Option.getOrElse(Option.scala:201)
  scala.cli.commands.shared.SharedOptions.bloopRifleConfig$$anonfun$1(SharedOptions.scala:544)
  scala.build.EitherCps$Helper.apply(EitherCps.scala:19)
  scala.cli.commands.shared.SharedOptions.bloopRifleConfig(SharedOptions.scala:554)
  scala.cli.commands.shared.SharedOptions.compilerMaker(SharedOptions.scala:563)
  scala.cli.commands.run.Run$.runCommand(Run.scala:145)
  scala.cli.commands.default.Default.runCommand(Default.scala:61)
  scala.cli.commands.default.Default.runCommand(Default.scala:40)
  scala.cli.commands.ScalaCommand.run(ScalaCommand.scala:368)
  scala.cli.commands.ScalaCommand.run(ScalaCommand.scala:350)
  caseapp.core.app.CaseApp.main(CaseApp.scala:157)
  scala.cli.commands.ScalaCommand.main(ScalaCommand.scala:335)
  caseapp.core.app.CommandsEntryPoint.main(CommandsEntryPoint.scala:166)
  scala.cli.ScalaCliCommands.main(ScalaCliCommands.scala:125)
  scala.cli.ScalaCli$.main0(ScalaCli.scala:276)
  scala.cli.ScalaCli$.main(ScalaCli.scala:108)
  scala.cli.ScalaCli.main(ScalaCli.scala)
com.github.plokhotnyuk.jsoniter_scala.core.JsonReaderException: unexpected end of input, offset: 0x00060000, buf:
+----------+-------------------------------------------------+------------------+
|          |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f | 0123456789abcdef |
+----------+-------------------------------------------------+------------------+
| 0005ffe0 | 77 6e 6c 6f 61 64 2f 32 30 2b 33 37 2f 62 65 6c | wnload/20+37/bel |
| 0005fff0 | 6c 73 6f 66 74 2d 6a 64 6b 32 30 2b 33 37 2d 6d | lsoft-jdk20+37-m |
+----------+-------------------------------------------------+------------------+
```

**Expected behaviour**
normal use

**Comment by Gedochao** [comment]
@fawdlstty do you have JVM 17+ set up on your local machine? 
JVM 17+ is a requirement for Scala CLI.
If it's not found locally, Scala CLI will try to download it, and inevitably fail if the download is impossible.
Are you working in an isolated/corporate environment?
If that is the case, please make sure to prepare the JVM locally. 
Additional setup may be necessary to either make Scala CLI be able to download dependencies in your environment or be able to work offline.

TL;DR details on your local setup are necessary for context.

**Comment by fawdlstty** [comment]
I can't find JVM software. Does jvm means jdk? If so, then I tried 17, 22, and the scala-cli didn't work

**Comment by Gedochao** [comment]
> I can't find JVM software. Does jvm means jdk?

JVM means Java Virtual Machine.
Yes, you need to have the JDK (Java Development Kit) for Java 17+ installed locally (either that, or let Scala CLI download it itself).

> If so, then I tried 17, 22, and the scala-cli didn't work

@fawdlstty What does your `JAVA_HOME` environment variable point to?
Did you get the same error as in the first post?
Any details on your machine? You are running Windows, correct?

Also, is there any reason why Scala CLI wouldn't be able to download the JVM itself on your machine? 
Isolated environments may need extra setup, so that is necessary context for me to be able to help you.

**Comment by Gedochao** [comment]
Can't reproduce, feel free to reopen if reproduction steps are provided.

**Comment by 2000ANNU** [comment]
@Gedochao I have the same problem i use VM Ubuntu

**Comment by fawdlstty** [comment]
@2000ANNU This is a network question, you need to try switching networks (delete `.scala-build` folder、switch VPN or proxy、move PC to other network、etc.) and retry.

**Comment by liuyuxivapor** [comment]
It seems not a network failure. The error directed to an empty url named `https://github.com/coursier/jvm-index/raw/master/index.json` that might be changed in the repo. 

Do you mean [this new url](https://github.com/coursier/jvm-index/blob/master/index.json) instead?

**Comment by fawdlstty** [comment]
@liuyuxivapor But I did solve the problem through the above steps. 佬跑通了chisel的话浇浇
